seismosocialdistancing.py

```diff
@@ -0,0 +1,248 @@
+            dRMS={}
+            for fmin, fmax in freqs:
+                ix = np.where((f<=fmax) & (f>=fmin))
+                # Parseval: the RMS in time domain is the sqrt of the integral of the power spectrum
+                rms = np.sqrt(np.trapz(damp[ix], 
+                                       f[ix]))
+                frange = "%.1f-%.1f"%(fmin, fmax)
+                if rms>0:
+                    dRMS[frange] = rms
+            if len(list(dRMS.keys())):
+                displacement_RMS[mseedid].append(dRMS)
+                times[mseedid].append(time)
+        self.displacement_RMS={}
+        for mseedid in self.mseedids:
+            index = pd.DatetimeIndex(times[mseedid])
+            self.displacement_RMS[mseedid] = pd.DataFrame(displacement_RMS[mseedid], 
+                                                          index=index)
+
+
+def sqlx2drms(sshuserhost,
+              network = 'CH',
+              station = 'SGEV',
+              location = '',
+              channel = 'HGZ,HGE,HGN',
+              dbname = 'AllNetworks',
+              start = UTCDateTime()-3*24*60*60,#"2020-03-07")
+              end = UTCDateTime(),# means "now"
+              freqs = [(0.1,1.0),(1.0,20.0),(4.0,14.0),(4.0,20.0)]):
+    """
+    Get PSDs from PQLX
+    
+    :type sshuserhost: string.
+    :param sshuserhost: ssh connection string, e.g. login@hostname.
+    :type network,station,location,channel: string.
+    :param network,station,location,channel: the mseed codes, use ',' as separator to get several channels.
+    :type start, end: `obspy.UTCDateTime``.
+    :param start, end: time window.
+    :type freqs: list of tuples.
+    :param freqs: frequency ranges (one each tuple).
+    :return: `PSDs`object.
+
+    .. rubric:: Basic Usage
+
+    You may omit everyhting but sshuserhost.
+
+    >>>myPSDs = sqlx2drms('login@hostname')
+    """
+    psds=PSDs()
+    datesplit = pd.date_range(start.datetime, 
+                              end.datetime, 
+                              freq="9D")
+    #print(datesplit)
+    for date in datesplit:
+        if date>= end.datetime:
+            break
+        ssh = subprocess.Popen(["ssh", 
+                                "-i .ssh/id_rsa", 
+                                sshuserhost],#sys.argv[1]],
+                               stdin =subprocess.PIPE,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               universal_newlines=True,
+                               bufsize=0)
+         
+        datelist = pd.date_range(date, 
+                                 date+pd.Timedelta(9, unit='D'), 
+                                 freq="30min")
+        #print(datelist)
+        # Send ssh commands to stdin
+        for date1 in datelist:
+            if date1>= end.datetime:
+                break
+            date2 = date1+pd.Timedelta(minutes=30)
+            for n in network.split(','):
```

Comment:
``for (n,s,l,c) in zip(network.split(","), station.split(","), location.split(","), channel.split(",")):``

works ?

Reply:
No. It does:
```

network='0'
station='A,B'
location=''
channel='Ω,µ'
for (n,s,l,c) in zip(network.split(","), station.split(","), location.split(","), channel.split(",")):
    print('%s.%s.%s.%s'%(n,s,l,c))
> 0.A..Ω
```
I want:
```
> 0.A..Ω
> 0.B..Ω
> 0.A..µ
> 0.B..µ
```

Reply:
```python
net = network.split(",")
sta = station.split(",")
loc = location.split(",")
chan = channel.split(",")
from itertools import combination_with_replacement
for (n,s,l,c) in combination(net,sta,loc,chan):
   # do something
```

or something similar :-) anyway, not important now

Reply:
OK next one :)